build-system/tasks/extension-generator/extension-doc.template.md

```diff
@@ -79,6 +154,114 @@ FILL THIS IN. Does this extension allow for properties to configure?
   </tr>
 </table>
 
+### optional-attribute-name (optional)
+
+Here, I write what optional-attribute-name will do to amp-component-name.
+
+[example preview=”top-frame” playground=”true”]
+
+```html
+<head>
+     <amp-component-name script tag>
+</head>
+<body>
+     <amp-component-name optional-attribute-name>
+          I am an inline executable code sample of the amp-component that demonstrates how
+          `optional-attribute-name` behaves.
+     </amp-component-name>
+</body>
+```
+
+[/example]
+
+Actions
+
+### action-name
+
+Description of action. Use cases of action-name. Include all the nuances, such as: amp-component-name needs to be identified with an `id` to work.
+
+[example preview=”top-frame” playground=”true”]
+
+```html
+<head>
+     <amp-component-name script tag>
+</head>
+<body>
+     <amp-component-name required-attribute>
+          Hello World!
+     </amp-component-name>
+     <button on="tap:amp-component-name.action-name">
+         Do Action
+    </button>
+</body>
+```
+
+[/example]
+
+## Events
+
+### event-name
+
+Description of event. Use cases of event-name. Include all the nuances, such as: amp-component-name needs to be identified with an `id` to work.
+
+[example preview=”top-frame” playground=”true”]
+
+```html
+<head>
+     <amp-component-name script tag>
+</head>
+<body>
+     <amp-component-name required-attribute
+       on="event-name: my-button.show">
+          Hello World!
+     </amp-component-name>
+     <button id="my-button" hidden>
+         Here I am!
+    </button>
+</body>
+```
+
+[/example]
+
+## Styling
+
+Explain how to style the element.
```

Comment:
Should every component have this section? Otherwise mention that this is optional.